src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp

```diff
@@ -370,28 +370,33 @@ namespace OpenMS
   {
     
     double ratio = 0.0;
-    if (component_1.metaValueExists(feature_name) && component_2.metaValueExists(feature_name))
-    {
-      double feature_1 = component_1.getMetaValue(feature_name);
-      double feature_2 = component_2.getMetaValue(feature_name);
-      ratio = feature_1/feature_2;
-      
-    }
-    else if (feature_name == "intensity")
+    if (feature_name == "intensity" && component_1.metaValueExists("native_id") && component_2.metaValueExists("native_id"))
     {
       const double feature_1 = component_1.getIntensity();
       const double feature_2 = component_2.getIntensity();
       ratio = feature_1 / feature_2;
     }
+    else if (feature_name == "intensity" && component_1.metaValueExists("native_id"))
+    {
+      LOG_DEBUG << "Warning: no IS found for component " << component_1.getMetaValue("native_id") << ".";
+      const double feature_1 = component_1.getIntensity();
+      ratio = feature_1;
+    }
+    else if (component_1.metaValueExists(feature_name) && component_2.metaValueExists(feature_name))
```

Comment:
these cases are bit difficult to follow.
Does this also cover the case were feature="intensity"? Then I think it would be wrong to request the metaValue("intensity")

Reply:
Essentially some feature_names are stored as metaValues and others are given members in the Feature class.  Specifically, I am trying to cover the case where one wants to quantify or filter based on peak area (i.e., "intensity") which is accessed via getIntensity() or peak height which is accessed by getMetaValue("peak_apex"int) for both the case where the first and second component exists or where the first component exists but the second does not.

Reply:
Ok that's how I understood it. But this will e.g. fail if feature_name = intensity but no native_id meta value exists, right?
Then  else if (component_1.metaValueExists(feature_name)) will be called with feature_name = intensity which will result in an error.
I think it would be better to distinguish two cases:
if (feature_name == "intensity")
{ // access member
}
else // feature name refers to meta value 
{ // access meta values
} 
and put the corresponding code in that parts.

Reply:
If there is no metavalue called "native_id" it will fail not work.  I am testing to see if the feature is actually populated with values.  Is there a way you would recommend testing to see if the feature has an intensity value?  That would avoid the need to test for the "native_id".

Reply:
Intensity is initialized to 0. Guess you could error out if this is zero and feature_name = intensity